File: internal/tools/colors.go
package tools

import (
	"os"
	"strings"
)

const (
	colorReset   = "\033[0m"
	colorMuted   = "\033[38;5;245m"
	colorSuccess = "\033[38;5;40m"
	colorError   = "\033[38;5;203m"
	colorTool    = "\033[38;5;214m\033[1m"
	colorFile    = "\033[38;5;75m"
)

func colorize(code, text string) string {
	if os.Getenv("NO_COLOR") != "" {
		return text
	}
	return code + text + colorReset
}

func muted(text string) string {
	return colorize(colorMuted, text)
}

func success(text string) string {
	return colorize(colorSuccess, text)
}

func failure(text string) string {
	return colorize(colorError, text)
}

func toolName(text string) string {
	return colorize(colorTool, text)
}

func fileName(text string) string {
	return colorize(colorFile, text)
}

func truncate(text string, max int) string {
	if len(text) <= max {
		return text
	}
	return strings.TrimSpace(text[:max]) + "..."
}

File: kai/tools/colors.py
import os

_NO_COLOR = os.environ.get("NO_COLOR") == "1"


def _markup(tag: str, text: str) -> str:
    if _NO_COLOR:
        return text
    return f"[{tag}]{text}[/]"


def muted(text: str) -> str:
    return _markup("dim", text)


def success(text: str) -> str:
    return _markup("green", text)


def failure(text: str) -> str:
    return _markup("red", text)


def tool_name(text: str) -> str:
    return _markup("bold orange3", text)


def file_name(text: str) -> str:
    return _markup("blue", text)


def truncate(text: str, max_len: int) -> str:
    if len(text) <= max_len:
        return text
    return text[:max_len].strip() + "..."
